Gems/EditorPythonBindings/Code/Source/ActionManager/PythonEditorAction.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/std/function/function_template.h>
+
+#include <Source/PythonCommon.h>
+#include <pybind11/pybind11.h>
+
+class PythonEditorAction
+{
+public:
+    AZ_TYPE_INFO(PythonEditorAction, "{1A5676D2-767B-4C2F-BC35-9CDDCE1430BB}");
+    AZ_CLASS_ALLOCATOR(PythonEditorAction, AZ::SystemAllocator, 0);
+
+    explicit PythonEditorAction(PyObject* handler);
+    PythonEditorAction(const PythonEditorAction& obj);
```

Comment:
suggestion: If we're defining the copy constructor it's usually good practice to also provide the assignment operator (and then implement or `=delete` the move constructor and assignment operator too)